astropy/io/fits/tests/test_hdulist.py

```diff
@@ -982,3 +982,15 @@ def test_pop_with_lazy_load(self):
         assert len(hdul3) == 2
         assert hdul3[0].header == hdu2.header
         assert hdul3[1].header == hdu.header
+
+    def test_pop_extname(self):
+        hdul = fits.open(self.data('o4sp040b0_raw.fits'))
+        assert len(hdul) == 7
+        hdu1 = hdul[1]
+        hdu4 = hdul[4]
+        hdu_popped = hdul.pop(('SCI', 2))
+        assert len(hdul) == 6
+        assert hdu_popped == hdu4
```

Comment:
Use `is` to check that they are the same instance (not sure what `==` does for HDUs ? It probably also check id but this could change in the future).